src/components/lineChart/nationalPractitionerLineChart.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useMemo, useState } from 'react';
+import Highcharts, { SeriesAreaOptions } from 'highcharts';
+import HighchartsReact from 'highcharts-react-official';
+
+import {
+  TimeframeOption,
+  ChartTimeControls,
+} from '~/components/chartTimeControls';
+import { getFilteredValues } from '~/components/chartTimeControls/chartTimeControlUtils';
+
+import styles from './lineChart.module.scss';
+import { formatNumber } from '~/utils/formatNumber';
+import { formatDate } from '~/utils/formatDate';
+
+interface Value {
+  date: number;
+  value: number | undefined | null;
+  week: Week;
+}
+
+type Week = {
+  start: number;
+  end: number;
+};
+
+type NationalPractitionerLineChartProps = {
+  values: Value[];
+  text: string;
+  timeframeOptions?: TimeframeOption[];
+};
+
+function getOptions(values: Value[]): Highcharts.Options {
+  const multipleAverageValues = values.length > 1;
+
+  const weeklyMeasurements: Week[] = values.map((value) => value.week);
+
+  const series: SeriesAreaOptions[] = [
+    {
+      type: 'area',
+      data: values.map((value) => [value.date, value.value]),
+      name: '',
+      showInLegend: false,
+      color: '#3391CC',
+      fillColor: 'rgba(51, 145, 204, 0.2)',
+      allowPointSelect: false,
+      marker: {
+        symbol: 'circle',
+        enabled: !multipleAverageValues,
+      },
+      events: {
+        legendItemClick: function () {
+          return false;
+        },
+      },
+      states: {
+        inactive: {
+          opacity: 1,
+        },
+      },
+    },
+  ];
+
+  const options: Highcharts.Options = {
+    chart: {
+      alignTicks: true,
+      animation: false,
+      backgroundColor: 'transparent',
+      borderColor: '#000',
+      borderRadius: 0,
+      borderWidth: 0,
+      colorCount: 10,
+      displayErrors: true,
+      height: 225,
+    },
+    legend: {
+      itemWidth: 300,
+      reversed: true,
+      itemHoverStyle: {
+        color: '#666',
+      },
+      itemStyle: {
+        color: '#666',
+        cursor: 'pointer',
+        fontSize: '12px',
+        fontWeight: 'normal',
+        textOverflow: 'ellipsis',
+      },
+    },
+    credits: {
+      enabled: false,
+    },
+    xAxis: {
+      lineColor: '#C4C4C4',
+      gridLineColor: '#ca005d',
+      type: 'datetime',
+      accessibility: {
+        rangeDescription: 'Verloop van tijd',
+      },
+      title: {
+        text: null,
+      },
+      categories: values.map((value) => value?.date.toString()),
+      labels: {
+        align: 'right',
+        // types say `rotation` needs to be a number,
+        // but that doesn’t work.
+        rotation: '0' as any,
+        formatter: function (): string {
+          if (this.isFirst || this.isLast) {
+            return formatDate(this.value * 1000, 'axis');
+          }
+          return '';
+        },
```

Comment:
I don't think a return type definition should be required here. Also an arrow function would be nice for inline. But I guess you got this code from somewhere else.

 ```suggestion
        formatter: () =>
          this.isFirst || this.isLast
            ? formatDate(this.value * 1000, 'axis')
            : '',
   
```

Reply:
Oops this won't work like that because you don't get the same "this" in the fat arrow context. So keep it a normal function declaration, but you can still do this:

```
formatter: function () {
         return this.isFirst || this.isLast
           ? formatDate(this.value * 1000, 'axis')
           : '' },
```

Reply:
I applied a refactor in the other places where this syntax was used